### DISCUSSION TITLE
Using external Midi devices

### DISCUSSION THREAD

**Discussion by Sheffdrum** [hydrogen-music/hydrogen#1623] [Hydrogen on Linux]
I’m having trouble trying to control Hydrogen via an external MIDI device.

My OS is Linux Mint Mate with a low latency kernel.

I’ve connected a Midi devices to Hydrogen by Qjackctl. 

I went to Hydrogen/Preferences/Midi system

I set 'Input' to the Midi device [the name of the device does not appear just a name that indicates that a Midi cable has been identified]. 

I set 'Output' to 'None' [I've tried all other options but none of them work]

'Channels' is set to 'All'.

The little blue ‘MIDI IN’ LED flashes when I press a key but there’s no sound.

The Hydrogen FAQ page states that - 

‘Keep in mind that Hydrogen uses a fixed midi-mapping (more info in the manual) so if your controller is sending a note that does not match with an instrument in your drumkit, you will hear no sound.’

FAQ advises changing the Midi mapping on the Midi device but I’ve tried to connect 2 x external Midi devices [not at the same time!] and it’s the same result with both. One device is an old Yamaha synth which works perfectly with other Linux Midi apps such as Rosegarden and Yoshimi.

(I’ve downloaded KMidimon but I don’t really understand how to use it.)

I’m thinking that external Midi devices have to be configured in some other way within Hydrogen and that’s why I’m hearing no sound.

(The external Midi devices that I’ve used are a Yamaha PSS-480 synth and a ‘Sheffield’ E Drum Kit.)

Any ideas?

**Comment by Sheffdrum** [comment]
I'd omitted to to connect the external midi device to hydrogen by 

qjackctl/setup/advanced

I changed the 'input' from the audio interface that I normally have plugged in to the external midi device and then everything worked as it should.

**Comment by theGreatWhiteShark** [comment]
> The little blue ‘MIDI IN’ LED flashes when I press a key but there’s no sound.

That sounds quite strange and like a bug. If MIDI IN LED flashes the MIDI message successfully reaches Hydrogen. From that point on things most likely go wrong within Hydrogen itself and not in your MIDI setup. It would be great if we could reproduce this issue in order to fix it.

> I'd omitted to to connect the external midi device to hydrogen by
>
> qjackctl/setup/advanced

Do you refer to the "Setup" dialog of QJackCtl? Are you altering the "Output Device" and "Input Device" combo boxes?

> I’ve connected a Midi devices to Hydrogen by Qjackctl.

Did you use the ALSA tab of QJackCtl's "Connections" dialog? For me this works just as setting the connection in the Preferences (which I wasn't aware of till now :D ).

> I changed the 'input' from the audio interface that I normally have plugged in to the external midi device and then everything worked as it should.

Glad to hear!

**Reply by Sheffdrum** [reply]
Many thanks for your reply

I forgot to select the external midi device by going to

qjackctl/setup/settings/advance

and selecting the external midi device in the input and output

when i did that it worked.

The thing is if the external midi device has not been selected in the ‘input’ of of qjackctl/setup/settings/advance the external midi device nevertheless shows up in the alsa tab of qjackctl ‘connect’ and if hydrogen and the external midi device are connected in the alsa tab the blue led meter in hydrogen flashes when a key is pressed on the external device but there will be no sound.

a stupid mistake on my part.
